### DISCUSSION TITLE
Support default comparison of structs

### DISCUSSION THREAD

**Discussion by hdante** [vlang/v#11864] [Ideas]
$ v
    Welcome to the V REPL (for help with V itself, type `exit`, then run `v help`).
    V 0.2.2 8547762
    Use Ctrl-C or `exit` to exit, or `help` to see other available commands
    >>> struct Struct1 {
    ...  field1 int
    ...  field2 string
    ...  field3 f32
    ...  field4 byte
    ... }
    >>> a := Struct1{}
    >>> b := Struct1{}
    >>> a < b
    error: undefined operation `Struct1` < `Struct1`
       20 | 
       21 | }
       22 | println(a < b)
          |         ~~~~~

Struct comparison might be implemented by default as:

    a < b <=> a.field1 < b.field1 ||                                                                        \
              (a.field1 == b.field1 && a.field2 < b.field2) ||                                              \
              (a.field1 == b.field1 && a.field2 == b.field2 && a.field3 < b.field3) ||                      \
              (a.field1 == b.field1 && a.field2 == b.field2 && a.field3 == b.field3 && a.field4 < b.field4)

**Comment by Gladear** [comment]
I highly doubt that this feature will make it to the language, especially given that operators can be overridden already

**Comment by mcastorina** [comment]
Adding another datapoint - Rust allows you to `derive` the traits necessary for sorting a struct.

> When derived on structs, it will produce a lexicographic ordering based on the top-to-bottom declaration order of the struct’s members.
